src/smc-webapp/project_store.coffee

```diff
@@ -181,6 +181,9 @@ class ProjectActions extends Actions
                 @redux.getActions('file_use')?.mark_file(@project_id, path, 'open')
                 @push_state('files/' + path)
                 @set_current_path(misc.path_split(path).head)
+                if redux.getStore('projects').get_my_group(@project_id) != store.open_files.getIn([path, 'component']).is_public
```

Comment:
This line of code **can't possibly be correct** because the left hand side is a string such as 'owner', 'collaborator', 'public' , and the right hand side is true or false (or undefined).